packages/trader/src/App/Components/Elements/Modals/ServicesErrorModal/__tests__/services-error-modal.spec.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import ServicesErrorModal from '../services-error-modal';
+
+jest.mock('../authorization-required-modal.jsx', () => jest.fn(() => 'AuthorizationRequiredModal'));
+jest.mock('../insufficient-balance-modal.jsx', () => jest.fn(() => 'InsufficientBalanceModal'));
+jest.mock('../company-wide-limit-exceeded-modal.jsx', () => jest.fn(() => 'CompanyWideLimitExceededModal'));
+jest.mock('@deriv/components', () => {
+    const original_module = jest.requireActual('@deriv/components');
+    return {
+        ...original_module,
+        Modal: jest.fn(() => <div data-testid='dt_mocked_modal' />),
+    };
+});
+
+describe('ServicesErrorModel component test cases', () => {
+    test('Should return null if code or message is missing', () => {
+        const onConfirm = jest.fn();
+        const services_error_mock = {
+            code: '',
+            message: '',
+            type: '',
+        };
+        const { container } = render(
+            <ServicesErrorModal
+                services_error={services_error_mock}
+                is_logged_in
+                is_virtual
+                is_visible
+                onConfirm={onConfirm}
+            />
+        );
+        expect(container).toBeEmptyDOMElement();
+    });
+
+    test('AuthorizationRequiredModal should render when code is AuthorizationRequired', () => {
+        const services_error_mock = {
+            code: 'AuthorizationRequired',
+            message: 'AuthorizationRequired',
+        };
+        const authorization_props = {
+            is_logged_in: true,
+            is_visible: true,
+            onConfirm: jest.fn(),
+        };
+        render(<ServicesErrorModal services_error={services_error_mock} {...authorization_props} />);
+        expect(screen.getByText('AuthorizationRequiredModal')).toBeInTheDocument();
+    });
+
+    test('InsufficientBalanceModal should render when code is InsufficientBalance', () => {
+        const services_error_mock = {
+            code: 'InsufficientBalance',
+            message: 'InsufficientBalance',
+        };
+        const insufficient_balance_props = {
```

Comment:
I suggest to remove `is_logged_in, is_visible, is_virtual ` and `onConfirm` outside `test` cases and put in below `describe`. That will help reuse code as the only prop that changes is `services_error_mock`.

Reply:
Will do